drivers/i3c/i3c_npcx.c

```diff
@@ -0,0 +1,2055 @@
+	.configure = npcx_i3c_configure,
+	.config_get = npcx_i3c_config_get,
+
+	.recover_bus = npcx_i3c_recover_bus,
+
+	.do_daa = npcx_i3c_do_daa,
+	.do_ccc = npcx_i3c_do_ccc,
+
+	.i3c_device_find = npcx_i3c_device_find,
+
+	.i3c_xfers = npcx_i3c_transfer,
+
+#ifdef CONFIG_I3C_USE_IBI
+	.ibi_enable = npcx_i3c_ibi_enable,
+	.ibi_disable = npcx_i3c_ibi_disable,
+#endif
+};
+
+#define I3C_NPCX_DEVICE(id)                                                                        \
+	PINCTRL_DT_INST_DEFINE(id);                                                                \
+	static void npcx_i3c_config_func_##id(const struct device *dev);                           \
+	static struct i3c_device_desc npcx_i3c_device_array_##id[] = I3C_DEVICE_ARRAY_DT_INST(id); \
+	static struct i3c_i2c_device_desc npcx_i3c_i2c_device_array_##id[] =                       \
+		I3C_I2C_DEVICE_ARRAY_DT_INST(id);                                                  \
+	static const struct npcx_i3c_config npcx_i3c_config_##id = {                               \
+		.base = (struct i3c_reg *)DT_INST_REG_ADDR(id),                                    \
+		.clock_dev = DEVICE_DT_GET(NPCX_CLK_CTRL_NODE),                                    \
+		.reset = RESET_DT_SPEC_INST_GET(id),                                               \
+		.clock_subsys = NPCX_DT_CLK_CFG_ITEM_BY_NAME(id, mclkd),                           \
+		.ref_clk_subsys = NPCX_DT_CLK_CFG_ITEM_BY_NAME(id, apb4),                          \
+		.irq_config_func = npcx_i3c_config_func_##id,                                      \
+		.common.dev_list.i3c = npcx_i3c_device_array_##id,                                 \
+		.common.dev_list.num_i3c = ARRAY_SIZE(npcx_i3c_device_array_##id),                 \
+		.common.dev_list.i2c = npcx_i3c_i2c_device_array_##id,                             \
+		.common.dev_list.num_i2c = ARRAY_SIZE(npcx_i3c_i2c_device_array_##id),             \
+		.pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(id),                                      \
+		IF_ENABLED(CONFIG_I3C_NPCX_DMA, (                                                  \
+			.mdma_clk_subsys = NPCX_DT_CLK_CFG_ITEM_BY_IDX(id, 2),                     \
+		))                                                                                 \
+		IF_ENABLED(CONFIG_I3C_NPCX_DMA, (                                                  \
+			.mdma_base = (struct mdma_reg *)DT_INST_REG_ADDR_BY_IDX(id, 1),            \
+		))                                                                                 \
+	};                                                                                         \
+	static struct npcx_i3c_data npcx_i3c_data_##id = {                                         \
+		.common.ctrl_config.scl.i3c = DT_INST_PROP_OR(id, i3c_scl_hz, 0),                  \
+		.common.ctrl_config.scl.i2c = DT_INST_PROP_OR(id, i2c_scl_hz, 0),                  \
+		.clocks.i3c_od_scl_hz = DT_INST_PROP_OR(id, i3c_od_scl_hz, 0),                     \
```

Comment:
should this be in `struct npcx_i3c_config`? does not looks like you are changing it in runtime

if that's the case then you are left with the other two, I would consider saving those in the config structure as well and initializing the data copy at init time from the code, would save the flash imagine from having an entire copy of the data structure just to intiialize two fields at boot time and the rest with zero